src/current/v23.2/plpgsql.md

```diff
@@ -0,0 +1,417 @@
+  ab851eb8-51eb-4800-8000-00000000014f | amsterdam | amsterdam    | ae147ae1-47ae-4800-8000-000000000022 | bbbbbbbb-bbbb-4800-8000-00000000000b | 62025 Welch Alley             | 4092 Timothy Creek Apt. 39  | 2018-12-31 03:04:05 | 2019-01-02 03:04:05 |   32.00
+~~~
+
+### Report messages and handle exceptions
+
+Use the `RAISE` statement to print messages for status or error reporting.
+
+~~~ sql
+RAISE level 'message' [, expressions ]
+  [ USING option = 'expression' [, ... ] ];
+~~~
+
+- `level` is the message severity. Possible values are `DEBUG`, `LOG`, `NOTICE`, `INFO`, `WARNING`, and `EXCEPTION`. Specify `EXCEPTION` to raise an error that aborts the current transaction.
+- `message` is a message string to display.
+- `expressions` is an optional, comma-separated list of [expressions](https://www.postgresql.org/docs/16/plpgsql-expressions.html) that provide values to replace any `%` placed within the message string. The number of expressions must match the number of `%` placeholders.
+- `option` is a type of additional information to include. Possible values are `MESSAGE`, `DETAIL`, `HINT`, or `ERRCODE`. To specify `MESSAGE`, use the following alternate syntax: 
+
+	~~~ sql
+	RAISE level USING MESSAGE = 'message';
+	~~~
+
+- `expression` is an expression to display that corresponds to the specified `option`. If `ERRCODE` is the specified option, this must be a valid [SQLSTATE error code or name](https://www.postgresql.org/docs/16/errcodes-appendix.html).
+
+For example:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+CREATE OR REPLACE PROCEDURE raise_time() AS $$
+  BEGIN
+    RAISE NOTICE 'current timestamp: %', now()
+	USING HINT = 'Call this procedure again for a different result';
+  END
+  $$ LANGUAGE PLpgSQL;
+~~~
+
+{% include_cached copy-clipboard.html %}
+~~~
+CALL raise_time();
+~~~
+
+~~~
+NOTICE: current timestamp: 2024-01-05 23:09:08.0601+00
+HINT: Call this procedure again for a different result
+CALL
+~~~
+
+#### Write exception logic
+
+Use an `EXCEPTION` statement to catch and handle specified errors. Any valid [SQLSTATE error code or name](https://www.postgresql.org/docs/16/errcodes-appendix.html) can be specified, other than `query_canceled`, `assert_failure`, or Class 40 (transaction rollback) errors. If a specified error is caught, the exception handling statements are executed. Any unspecified errors are caught by `WHEN OTHERS`, if included.
+
+~~~ sql
+EXCEPTION
+	WHEN error THEN
+		handle_exception;
+	[ WHEN error_n THEN
+		handle_exception_n; ]
+	[ WHEN OTHERS THEN
+		handle_other_exceptions; ]
+~~~
+
+`EXCEPTION` logic is typically included after the main body of a PL/pgSQL block. For example:
+
+~~~ sql
+BEGIN
+	...
+  EXCEPTION
+    WHEN not_null_violation THEN
+      RETURN 'not_null_violation';
+    WHEN OTHERS THEN
+      RETURN 'others';
```

Comment:
The single quotes here aren't valid syntax. Only the SQLSTATE code needs quotes.